models/src/main/java/io/sphere/sdk/attributes/AttributeAccess.java

```diff
@@ -183,55 +183,64 @@ public boolean canHandle(final AttributeDefinition attributeDefinition) {
+                        final ReferenceType attributeType = (ReferenceType) attributeDefinition.getAttributeType();
+
+                        return attributeType.getReferenceTypeId().equals(referenceType.getReferenceTypeId());
+                    } else {
+                        return false;
                     }
-                    return canHandle;
                 });
     }
 
     private static <T> AttributeAccess<Set<T>> ofSet(final Class<? extends AttributeType> typeClass, final TypeReference<Set<T>> typeReference) {
         final AttributeMapper<Set<T>> mapper = AttributeMapper.of(typeReference);
         return new AttributeAccess<>(mapper, attributeDefinition -> {
-            boolean canHandle = false;
-            if (SetAttributeDefinition.class.isAssignableFrom(attributeDefinition.getClass())) {
-                final SetAttributeDefinition setAttributeDefinition = (SetAttributeDefinition) attributeDefinition;
-                final SetType attributeType = setAttributeDefinition.getAttributeType();
-                canHandle = typeClass.isAssignableFrom(attributeType.getElementType().getClass());
+            if (attributeDefinition.getAttributeType() instanceof SetType) {
+                final SetType attributeType = (SetType) attributeDefinition.getAttributeType();
+
+                return typeClass.isAssignableFrom(attributeType.getElementType().getClass());
+            } else {
+                return false;
             }
-            return canHandle;
         });
     }
 
     private static <T> AttributeAccess<Set<Reference<T>>> ofSet(final ReferenceType requiredReferenceType, final TypeReference<Set<Reference<T>>> typeReference) {
         final AttributeMapper<Set<Reference<T>>> mapper = AttributeMapper.of(typeReference);
         return new AttributeAccess<>(mapper, attributeDefinition -> {
-            boolean canHandle = false;
-            if (SetAttributeDefinition.class.isAssignableFrom(attributeDefinition.getClass())) {
-                final SetAttributeDefinition setAttributeDefinition = (SetAttributeDefinition) attributeDefinition;
-                final SetType attributeType = setAttributeDefinition.getAttributeType();
+            final boolean canHandle;
+
+            if (attributeDefinition.getAttributeType() instanceof SetType) {
+                final SetType attributeType = (SetType) attributeDefinition.getAttributeType();
+
                 if (attributeType.getElementType() instanceof ReferenceType) {
                     final ReferenceType referenceType = (ReferenceType) attributeType.getElementType();
+
                     canHandle = referenceType.getReferenceTypeId().equals(requiredReferenceType.getReferenceTypeId());
+                } else {
+                    canHandle = false;
                 }
+            } else {
+                canHandle = false;
             }
+
             return canHandle;
```

Comment:
yeah, I needed to change it because all of the subclasses of `AttributeDefinition` are gone now, and this method was using them